### ISSUE TITLE
A Congratulations you won text box?

### ISSUE THREAD

**Issue by turkishfilms** [trevorbayless/cli-chess#33]
### What feature would you like to see in cli-chess?

I won and it was great in Offline mode. But then it still prompted me for a move and the only way to exit was to type quit and then it pronounced the checkmated white side victorious. ;/

**Comment by trevorbayless** [comment]
If you checkmate your opponent (offline or online) it should immediately notify you of the checkmate, and end the game. If that didn't happen it sounds like there could be a bug. I'd be interested in seeing the logs if you still have them around.

The various log files can be found here: `~/.config/cli-chess/` and end in `.log`.

**Comment by turkishfilms** [comment]
[Log files](https://github.com/turkishfilms/cli-chess/blob/master/cli-chess.log)

**Comment by trevorbayless** [comment]
Thanks for supplying the logs! Below are the moves made in the game:

```d2d4 g8f6 e2e3 d7d5 f1d3 c7c5 c2c4 d5c4 d3c4 e7e6 c1d2 c5d4 c4e2 d4e3 d2e3 b7b5 e2b5 f8b4 b1c3 e8g8 a2a3 b4c3 b2c3 b8c6 e3g5 d8d1 e2d1 f6e4 g1f3 e4c3 h2h4 h7h6 g5c1 f8d8 d1c2 c6d4 e1d2 d4c2 d2c2 c3d5 c2b3 b7b6 a3a4 c8a6 a1b1 a8c8 b3b2 d5b4 f3e1 a6d3 g2g4 d3b1 e1g2 b1e4 b2a3 b4c2 a3a2 e4g2 h1g1 g2d5 a2b1 c2d4 c1d2 d4f3 d2a5 f3g1 a5e1 d5e4 b1a2 c8c2 a2a1 d8d1```

The moves in green are good and end with the engine checking you on `e2b5`. Somehow you were able to continue the game without acknowledging the check (red section)

```diff
+ d2d4 g8f6 e2e3 d7d5 f1d3 c7c5 c2c4 d5c4 d3c4 e7e6 c1d2 c5d4 c4e2 d4e3 d2e3 b7b5 e2b5
- f8b4 b1c3 e8g8 a2a3 b4c3 b2c3 b8c6 e3g5 d8d1 e2d1 f6e4 g1f3 e4c3 h2h4 h7h6 g5c1 f8d8 d1c2 c6d4 e1d2 d4c2 d2c2 c3d5 c2b3 b7b6 a3a4 c8a6 a1b1 a8c8 b3b2 d5b4 f3e1 a6d3 g2g4 d3b1 e1g2 b1e4 b2a3 b4c2 a3a2 e4g2 h1g1 g2d5 a2b1 c2d4 c1d2 d4f3 d2a5 f3g1 a5e1 d5e4 b1a2 c8c2 a2a1 d8d1
```

The position up to green looked like this for you:
![image](https://github.com/user-attachments/assets/915a1ffc-a052-464c-a846-893e243e74ec)



I have some follow up questions:

- Do you recall if you used the "takeback" feature to make a new move after the engine checked you, or at any other point in the game? (I realize now that I don't log these events, so this could explain how you were able to seemingly "continue on" without acknowledging the check - I'll need to update the logging to account for this).

- Did you have multiple instances of cli-chess open?

**Comment by turkishfilms** [comment]
I believe I did you take back in this game to see how it would work. I may have us
ed it around this time. I only had one instance running. I would be very interested to help you get to the bottom of this. Perhaps a
lso outputting a pgn of the game would be helpful. I'll look into your code more and may send a pr your way

**Comment by trevorbayless** [comment]
I've released a new version of cli-chess ([v1.4.2](https://github.com/trevorbayless/cli-chess/releases/tag/v1.4.2)) which from my testing addresses this issue. It also adds additional logging which should make root cause discovery next time a little quicker. Feel free to try out the new version and open a bug report if you run into this again, or any other issues!

As far as producing a PGN at the end of the game, it's been requested (https://github.com/trevorbayless/cli-chess/issues/19) and I'd happily review any PR that comes across for it!

I'm closing this enhancement as you should always see a proper game over message when a game completes. Hopefully the changes in v1.4.2 actually make that happen for you now!
